### PR TITLE
deploy apollo stack to the swarm

### PR DESCRIPTION
**Pull request by aibeb**
## What's the purpose of this PR

deploy apollo stack to the swarm

## Which issue(s) this PR fixes:

deploy apollo stack to the swarm

## Brief changelog

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [ ] Read the [Contributing Guide](https://github.com/ctripcorp/apollo/blob/master/CONTRIBUTING.md) before making this pull request.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit tests to verify the code.
- [ ] Run `mvn clean test` to make sure this pull request doesn't break anything.

### CODE REVIEW

File: scripts/apollo-on-swarm/README.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+docker service create \
+  --name apollo-configservice \
+  --replicas 1 \
+  --restart-condition on-failure \
+  --env SPRING_DATASOURCE_URL=jdbc:mysql://apollo-mysql:3306/ApolloConfigDB?characterEncoding=utf8 \
+  --env SPRING_DATASOURCE_USERNAME=root \
+  --env SPRING_DATASOURCE_PASSWORD= \
+  --env SPRING_PROFILES_ACTIVE=github,kubernetes \
+  --env APOLLO.CONFIG-SERVICE.URL=http://${APOLLO.CONFIG-SERVICE.URL}:8080 \
+  --env APOLLO.ADMIN-SERVICE.URL=http://apollo-adminservice:8090 \
```
Comment (nobodyiam) [inline scripts/apollo-on-swarm/README.md:60]: env 的命名规范一般都是用下划线的，这里统一也改一下吧
```suggestion
  --env APOLLO_CONFIG_SERVICE_URL=http://${APOLLO.CONFIG-SERVICE.URL}:8080 \
  --env APOLLO_ADMIN_SERVICE_URL=http://apollo-adminservice:8090 \
```

Comment (aibeb) [reply scripts/apollo-on-swarm/README.md:60]: 这里没有采用下划线是因为apollo-configservbice/src/main/config/application-github.properties中没有接受下划线环境变量的配置, 如果需要支持下划线需要pr去修改application-github.properties 这个我可以附带到我到pr中

Comment (nobodyiam) [reply scripts/apollo-on-swarm/README.md:60]: 下划线是支持的，spring 会自动适配不同格式的配置的，`apollo.config-service.url`和`APOLLO_CONFIG_SERVICE_URL`等价

File: scripts/apollo-on-swarm/README.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  --network apollo \
+  --publish 8070:8070 \
+  apolloconfig/apollo-portal:${VERSION}
+```
+
+## 通过docker-compose.yaml部署
+
+```shell
+# 注意文件中 APOLLO_CONFIG_SERVICE_HOST 更换为集群所处宿主机ip地址，保证外部客户端访问
+docker stack deploy -c ./docker-compose.yaml apollo
```
Comment (nobodyiam) [inline scripts/apollo-on-swarm/README.md:104]: mac 电脑上运行该命令报错

```bash
invalid interpolation format for services.apollo-mysql.volumes.[]: "$(pwd)/../scripts/docker-quick-start/sql:/docker-entrypoint-initdb.d". You may need to escape any $ with another $.
```

Comment (nobodyiam) [reply scripts/apollo-on-swarm/README.md:104]: 注释掉 apollo-mysql 相关内容后，看上去也是环境变量没拿到

```bash
Error response from daemon: rpc error: code = InvalidArgument desc = ContainerSpec: "apolloconfig/apollo-configservice:" is not a valid repository/tag
```

**Review by nobodyiam — CHANGES_REQUESTED** [review]

**Comment by JaredTan95** [comment]
as https://github.com/ctripcorp/apollo/pull/3747 merged, please update [changs log](https://github.com/ctripcorp/apollo/blob/master/CHANGES.md). :-)

**Comment by nobodyiam** [comment]
@aibeb any update on this?

**Pull request closed without merging**